lib/filterx/object-primitive.c

```diff
@@ -139,12 +139,54 @@ _double_add(FilterXObject *s, FilterXObject *object)
 }
 
 gboolean
-double_repr(double val, GString *repr)
+double_repr(double dbl, GString *repr)
 {
-  gsize init_len = repr->len;
-  g_string_set_size(repr, init_len + G_ASCII_DTOSTR_BUF_SIZE);
-  g_ascii_dtostr(repr->str + init_len, G_ASCII_DTOSTR_BUF_SIZE, val);
-  g_string_set_size(repr, init_len + strlen(repr->str + init_len));
+  if (isnan(dbl))
+    {
+      g_string_append_len(repr, "NaN", 3);
+    }
+  else if (isinf(dbl))
+    {
+      if (dbl < 0)
+        g_string_append_c(repr, '-');
+      g_string_append_len(repr, "Infinity", 8);
+    }
+  else
+    {
+      gchar buf[32];
+      gchar *dot;
+
+      snprintf(buf, sizeof(buf), "%.17g", dbl);
+
+      /* find fractions, either '.' or ',' can separate integer and fraction
+       * part */
+      dot = strchr(buf, '.');
+      if (!dot)
+        {
+          dot = strchr(buf, ',');
+          if (dot)
+            *dot = '.';
```

Comment:
`g_ascii_dtostr()` uses `"%.17g"` and has no locale problems (comma).

Reply:
I pushed the `g_ascii_dtostr()` version but the rebase made it disappear.

Reply:
sorry, for dropping your change, did not notice you pushed another version. let me redo that.